registry/api/errcode/errors.go

```diff
@@ -69,20 +69,28 @@ func (ec *ErrorCode) UnmarshalText(text []byte) error {
 // WithDetail creates a new Error struct based on the passed-in info and
 // set the Detail property appropriately
 func (ec ErrorCode) WithDetail(detail interface{}) Error {
```

Comment:
I mentioned it to @stevvooe today before looking at this PR but these ErrorCode.WithXxx methods seem inconsistent with other WithXxx methods in Go (particularly the context package, and our own auth package). It seems very odd to have them return a different type to the receiver.